irods/message/__init__.py

```diff
@@ -163,11 +173,25 @@ def _recv_message_in_len(sock, size):
             if getattr(e, 'winerror', 0) != 10045:
                 raise
             buf = sock.recv(size_left)
+
+        # Break out of the loop
+        if len(buf) == 0:
+            break
         size_left -= len(buf)
         if retbuf is None:
             retbuf = buf
         else:
             retbuf += buf
+
+    # This method is supposed to read and return 'size' bytes from the socket
+    # If it reads no bytes (retbuf will be None), or if it reads less number of
+    # bytes than 'size', must throw an exception
+    if retbuf is None or len(retbuf) != size:
```

Comment:
If the method did not read 'size' bytes from socket, something went wrong. Raise a scoket.error exception with a useful message.